src/wccp2.cc

```diff
@@ -553,17 +552,16 @@ static char
 wccp2_update_md5_security(char *password, char *ptr, char *packet, int len)
 {
     uint8_t md5Digest[SQUID_MD5_DIGEST_LENGTH];
-    char pwd[WCCP2_PASSWORD_LEN];
+    char pwd[WCCP2_PASSWORD_LEN + 1];
     SquidMD5_CTX M;
 
     struct wccp2_security_md5_t *ws;
 
     debugs(80, 5, "wccp2_update_md5_security: called");
 
     /* The password field, for the MD5 hash, needs to be 8 bytes and NUL padded. */
-    memset(pwd, 0, sizeof(pwd));
-    strncpy(pwd, password, sizeof(pwd));
-    pwd[sizeof(pwd) - 1] = '\0';
+    memset(pwd, 0, WCCP2_PASSWORD_LEN);
+    xstrncpy(pwd, password, WCCP2_PASSWORD_LEN);
```

Comment:
The sizeof(pwd) were correct. Please revert the change to size parameters for both these calls.

```suggestion
    memset(pwd, 0, sizeof(pwd));  // prevent WCCP leaking data from stack area
    xstrncpy(pwd, password, sizeof(pwd));
```

Reply:
Done.